ez-launch/sources/ez_launch.move

```diff
@@ -0,0 +1,413 @@
+
+        let i = 0;
+        let length = vector::length(&token_name_vec);
+        while (i < length) {
+            let token = token_pre_mint_internal(
+                creator,
+                collection,
+                *vector::borrow(&token_description_vec, i),
+                *vector::borrow(&token_name_vec, i),
+                *vector::borrow(&token_uri_vec, i),
+            );
+            vector::push_back(&mut collection_details_obj.to_be_minted, token);
+            i = i + 1;
+        };
+    }
+
+    public entry fun mint(
+        user: &signer,
+        collection: Object<Collection>,
+    ) acquires CollectionDetails, EZLaunchRefs {
+        mint_internal(user, collection);
+    }
+
+    public entry fun set_minting_status(creator: &signer, collection: Object<Collection>, ready_to_mint: bool) acquires CollectionDetails, EZLaunchRefs {
+        assert_owner<Collection>(signer::address_of(creator), collection);
+        let object_signer = authorized_collection_signer(creator, collection);
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+
+        collection_details_obj.ready_to_mint = ready_to_mint;
+    }
+        
+    // ================================= View  ================================= //
+
+    #[view]
+    public fun minting_ended(collection: Object<Collection>): bool acquires CollectionDetails, EZLaunchRefs {
+        let object_signer = collection_signer(collection);
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+
+        vector::length(&collection_details_obj.to_be_minted) == 0
+    }
+
+    // ================================= Helper  ================================= //
+
+    fun create_collection_internal(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        mutable_collection_metadata: bool,
+        mutable_token_metadata: bool,
+        random_mint: bool,
+        is_soulbound: bool,
+        tokens_burnable_by_creator: bool,
+        tokens_transferable_by_creator: bool,
+        max_supply: Option<u64>,
+        mint_fee: Option<u64>,
+        royalty_numerator: Option<u64>,
+        royalty_denominator: Option<u64>,
+    ): Object<Collection> acquires CollectionDetails {
+        let creator_addr = signer::address_of(creator);
+
+        assert!(option::is_some(&royalty_numerator) == option::is_some(&royalty_denominator), error::invalid_argument(EROYALTY_CONFIGURATION_INVALID));
+        let royalty = if (option::is_some(&royalty_numerator) && option::is_some(&royalty_denominator)) {
+            option::some(royalty::create(option::extract(&mut royalty_numerator), option::extract(&mut royalty_denominator), creator_addr))
```

Comment:
We need checks on if the royalty passed in is zero or not. This should not be allowed. Ideally, I rather pass in u64 rather than `Option`. Kevin also mentioned that there were problems passing in `Option` in the past.